cunumeric/array.py

```diff
@@ -1503,6 +1505,16 @@ def __setitem__(self, key, value):
         Set ``self[key]=value``.
 
         """
+        if self._broadcasted is True:
+            warnings.warn(
+                "This array is a broadcasted array from "
+                "cunumeric.broadcast_arrays. "
+                "Writing to this array will be prohibited "
+                "in the future release "
```

Comment:
Why don't we prohibit writing to the broadcasted array now?

Reply:
`numpy` allows the writing for the output of `broadcast_arrays` and `broadcast`. For `broadcast_arrays`, the documentation says the writing to the output will be disallowed in the future release. For 'broadcast', there's not any word about that. For cuNumeric, we can have our own policy regarding this. I keep this implementation to behave in the same way as numpy

Reply:
I see. We can keep having warning until it is deprecated in NumPy. Have you actually tried modifying broadcasted array? Can you add it to the tests?

Reply:
> I see. We can keep having warning until it is deprecated in NumPy. Have you actually tried modifying broadcasted array? Can you add it to the tests?

In `numpy`, the first attempt to write to the broadcasted array show the warning. It allows the attempt.  One thing I couldn't find regarding this is where the default warning level at cuNumeric is set.